src/interactive/AccordionPanelGroup.jsx

```diff
@@ -90,7 +118,9 @@ class AccordionPanelGroup extends React.Component {
 			>
 				{
 					this.renderAccordionPanels().map((panel, i) => (
```

Comment:
there's something funky happening here - `this.accordionPanels` is set to a _cloned_ list of panels in the constructor, and then `this.renderAccordionPanels` does a clone of the clone inside `render()` - you should only need to clone once, and then I think you can just use `this.accordionPanels` on this line instead of `this.renderAccordionPanels()`.

Related: the`clonePanel` method should _not_ be setting a `key` - it's not needed, since each individual panel is wrapped inside an `<li>` with a `key` prop here in the `render()` method.

I think the net render result will be the same, but you can simplify the implementation by stripping out a couple things.

Reply:
got it. yea I thought that was funky too but didnt question it enough.

Reply:
well I realize the clone is necessary to set the appropriate `isOpen` prop from state during render, unless there is another way. I refactored the code into `initPanel` and `cloneAccordionPanels` to be a little less awkward and hopefully more clear what is going on .